cincinnati/src/lib.rs

```diff
@@ -734,47 +729,134 @@ mod tests {
         graph
     }
 
-    pub(crate) fn generate_custom_graph(
-        start: usize,
-        count: usize,
-        mut metadata: HashMap<usize, HashMap<String, String>>,
-        edges: Option<Vec<(usize, usize)>>,
-    ) -> Graph {
-        let mut graph = Graph::default();
+    pub type TestMetadata = Vec<(usize, HashMap<String, String>)>;
+    pub type TestEdges = Vec<(usize, usize)>;
 
-        let nodes: Vec<daggy::NodeIndex> = (start..(start + count))
-            .map(|i| {
-                let metadata = metadata.remove(&i).unwrap_or_default();
+    #[derive(Debug, Clone)]
+    pub struct TestGraphBuilder {
+        image: String,
+        metadata: TestMetadata,
+        edges: Option<TestEdges>,
+        version_template: String,
+        enable_payload_suffix: bool,
+    }
 
-                let release = Release::Concrete(ConcreteRelease {
-                    version: format!("{}.0.0", i),
-                    payload: format!("image/{}.0.0", i),
-                    metadata,
-                });
-                graph.dag.add_node(release)
-            })
-            .collect();
+    impl Default for TestGraphBuilder {
+        fn default() -> Self {
+            TestGraphBuilder {
+                image: "image".to_string(),
+                metadata: Default::default(),
+                edges: None,
+                version_template: "{{i}}.0.0".to_string(),
+                enable_payload_suffix: false,
+            }
+        }
+    }
+
+    impl TestGraphBuilder {
+        pub fn new() -> Self {
+            Default::default()
+        }
 
-        assert_eq!(count as u64, graph.releases_count());
+        pub fn with_image(mut self, image: &str) -> Self {
+            self.image = image.to_string();
+            self
+        }
 
-        if let Some(edges) = edges {
-            for (key, value) in &edges {
-                let one = nodes[*key];
-                let two = nodes[*value];
-                graph.dag.add_edge(one, two, Empty {}).unwrap();
-            }
-            assert_eq!(edges.len(), graph.dag.edge_count());
-        } else {
-            for i in 0..(nodes.len() - 1) {
-                let one = nodes[i];
-                let two = nodes[i + 1];
-                graph.dag.add_edge(one, two, Empty {}).unwrap();
-            }
-        };
+        pub fn with_metadata(mut self, metadata: TestMetadata) -> Self {
+            self.metadata = metadata;
+            self
+        }
 
-        graph
+        pub fn with_edges(mut self, edges: Option<TestEdges>) -> Self {
+            self.edges = edges;
+            self
+        }
+
+        pub fn with_version_template(mut self, version_template: &str) -> Self {
+            self.version_template = version_template.to_string();
+            self
+        }
+
+        pub fn enable_payload_suffix(mut self, enable: bool) -> Self {
```

Comment:
It seems this doesn't get tested yet, right?

Reply:
Right, i's not used in any tests as of the commit it's being added in.